ext/standaloneusers/templates/CRM/Standaloneusers/Page/Login.tpl

```diff
@@ -26,45 +26,29 @@
 <script>
   document.addEventListener('DOMContentLoaded', () => {
 
-    const submitBtn = document.getElementById('loginSubmit'),
+    const form = document.getElementById('login-form'),
       username = document.getElementById('usernameInput'),
       password = document.getElementById('passwordInput');
 
-    submitBtn.addEventListener('click', async e => {
+    form.addEventListener('submit', async e => {
       e.preventDefault();
 
-      const response = await fetch(CRM.url("civicrm/authx/login"), {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/x-www-form-urlencoded'
-        },
-        //body: '_authx=Basic ' + btoa(encodeURIComponent(`${username.value}:${password.value}`))
-        body: '_authx=Basic ' + encodeURIComponent(btoa(`${username.value}:${password.value}`))
-      });
-      if (!response.ok) {
-        const contentType = response.headers.get("content-type");
-        let msg = 'Unexpected error';
-        if (!contentType || !contentType.includes("application/json")) {
-          // Non-JSON response; an error.
-          msg = await response.text();
-          // Example error string: 'HTTP 401 Invalid credential'
-          msg = msg.replace(/^HTTP \d{3} /, '');
+      let errorMsg = 'Unexpected error';
+      try {
+        const response = await CRM.api4('User', 'login', {
+          username: username.value,
+          password: password.value,
+        });
+        if (response.url) {
```

Comment:
It looks like we lose the ability to login straight to a page that isn't the dashboard. 

I think I'd rather have TOTP than that feature. But maybe the API could only return `response.url` for the secondary form, and use `location.reload` if the api call is successful and no response.url is returned?

Reply:
> But maybe the API could only return response.url for the secondary form, and use location.reload if the api call is successful and no response.url is returned?

Oops, no, cause we switch to the TFA url.

Can we pass the starting url through as an extra api param/put it in the session?